fs/dafsserver/dafsserver.cpp

```diff
@@ -5834,41 +5834,43 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
 
     void checkTimeout()
     {
-        if (msTick()-clientcounttick>1000*60*60)
         {
             CriticalBlock block(ClientCountSect);
-            if (TF_TRACE_CLIENT_STATS && (ClientCount || MaxClientCount))
-                PROGLOG("Client count = %d, max = %d", ClientCount, MaxClientCount);
-            clientcounttick = msTick();
-            MaxClientCount = ClientCount;
-            if (closedclients)
+            if (msTick() - clientcounttick > 1000 * 60 * 60)
```

Comment:
the write to clientcounttick was tested outside the crit, and altered in it, but it didn't need protecting. ClientCountSect is really only protecting concurrency to ClientCount and MaxClientCount.
It would be better to move the clientcounttick, to just after the if, and scope this crit around the ClitnCount/MaxClientCount code, i.e.:
```
            clientcounttick = msTick();
            {
                CriticalBlock block(ClientCountSect);
                if (TF_TRACE_CLIENT_STATS && (ClientCount || MaxClientCount))
                    PROGLOG("Client count = %d, max = %d", ClientCount, MaxClientCount);
                MaxClientCount = ClientCount;
            }
```

Reply:
It would also be worth noting somewhere that clientcounttick and closedclients are members only touched/read by checkTimeout() that is not contended itself.